osi_vehicle.proto

```diff
@@ -0,0 +1,829 @@
+syntax = "proto2";
+
+option optimize_for = SPEED;
+
+import "osi_version.proto";
+import "osi_common.proto";
+import "osi_object.proto";
+import "osi_occupant.proto";
+
+package osi3;
+
+//
+// \brief The vehicle-message is a deeper description of a vehicle.
+// Consists of different messages categorizing the vehicle in:
+// Vehicle-Basics, Vehicle-Kinematics, Vehicle-Powertrain, Vehicle-SteeringWheel, Vehicle-Wheels,
+// Vehicle-Light-State, Vehicle-Automated-Driving-Functions.
+// As it is an description of the whole vehicle (focus on cars) it can be used as interface for various reasons, 
+// e.g. to the vehicle-model for kinematics calculation or even for the visualisation in a graphic-engine.
+//
+message Vehicle
+{
+    //
+    // The interface version used by the sender (i.e. the simulation
+    // environment).
+    //
+    optional InterfaceVersion version = 1;
+    
+    //
+    // The data timestamp of the simulation environment. The zero time point is
+    // arbitrary but must be identical for all messages.
+    // Recommendation: Zero time point for start point of the simulation.
+    //
+    optional Timestamp timestamp = 2;
+    
+    //
+    // The base parameters of the vehicle.
+    //
+    optional VehicleBasics vehicle_basics = 3;
+    
+    //
+    // This is the interface, that describes how the vehicle is moving.
+    // All coordinates and orientations are relative to the global ground truth
+    // coordinate system.
+    //
+    optional VehicleKinematics vehicle_kinematics = 4;
+    
+    //
+    // Interface regarding the powertrain.
+    //
+    optional VehiclePowertrain vehicle_powertrain = 5;
+    
+    //
+    // Interface regarding the SteeringWheel.
+    // The focus here is on the steering wheel.
+    //
+    optional VehicleSteeringWheel vehicle_steering_wheel = 6;
+    
+    //
+    // Interface regarding the wheels.
+    // The focus here is on the physical description of a wheel.
+    //
+    optional VehicleWheels vehicle_wheels = 7;
+    
+    //
+    // Interface regarding the light.
+    // The focus here is on the states of the lights.
+    //
+    optional MovingObject.VehicleClassification.LightState vehicle_light_state = 8;
+    
+    //
+    // Interface regarding the navigation.
+    //
+    optional VehicleLocalization vehicle_localization = 9;
+    
+    //
+    // Interface regarding automated-driving functions.
+    // The focus here is on states and intentions of a function.
+    //
+    repeated VehicleAutomatedDrivingFunction vehicle_automated_driving_function = 10;
+
+    //
+    // \brief The absolut base parameters of the vehicle.
+    //
+    message VehicleBasics
+    {
+        // The id of the car in the simulation. 
+        //
+        optional Identifier id = 1;
+
+        // The brand of the car. 
+        //
+        // \note It is implementation-specific how the string is filled.
+        //
+        optional string brand = 2;
```

Comment:
brand and model name may better be a part of the reference_string. If the specific model is not available one should use a generic type like the types in MovingObejct

Reply:
I deleted brand and modelname. So far it is not really necessary.